server/src/main/java/org/spine3/server/entity/FieldMasks.java

```diff
@@ -0,0 +1,182 @@
+ */
+
+package org.spine3.server.entity;
+
+import com.google.protobuf.Descriptors;
+import com.google.protobuf.FieldMask;
+import com.google.protobuf.Message;
+import com.google.protobuf.ProtocolStringList;
+import org.spine3.protobuf.KnownTypes;
+import org.spine3.protobuf.TypeUrl;
+
+import javax.annotation.Nullable;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * <p>Util class for filtering fields that are included in an instance of {@link Message} or collection of {@link Message}s.
+ * Provides basic functionality for processing of e.g. a {@code org.spine3.client.Query} to in-memory storage.</p>
+ *
+ * @author Dmytro Dashenkov
+ */
+@SuppressWarnings("UtilityClass")
+public class FieldMasks {
+
+    private FieldMasks() {
+    }
+
+    /**
+     * <p>Applies given {@code FieldMask} to givem collection of {@link Message}s.
+     * Does not change the {@link Collection} itself.</p>
+     *
+     * <p>The {@code FieldMask} must be valid for this operation.</p>
+     *
+     * @param mask     {@code FieldMask} to apply to each item of the input {@link Collection}.
+     * @param entities {@link Message}s to filter.
+     * @param typeUrl  Type of the {@link Message}s.
+     * @return Non-null unmodifiable {@link Collection} of {@link Message}s of the same type that the input had.
+     * @see #isValid(FieldMask)
+     */
+    @SuppressWarnings({"MethodWithMultipleLoops", "unchecked"})
+    public static <M extends  Message, B extends Message.Builder> Collection<M> applyMask(@SuppressWarnings("TypeMayBeWeakened") FieldMask mask, Collection<M> entities, TypeUrl typeUrl) {
+        final List<M> filtered = new ArrayList<>();
+        final ProtocolStringList filter = mask.getPathsList();
+
+        final Class<B> builderClass = getBuilderForType(typeUrl);
+
+        if (filter.isEmpty() || builderClass == null) {
+            return Collections.unmodifiableCollection(entities);
+        }
+
+        try {
+            final Constructor<B> builderConstructor = builderClass.getDeclaredConstructor();
+            builderConstructor.setAccessible(true);
+
+            for (Message wholeMessage : entities) {
+                final B builder = builderConstructor.newInstance();
+
+                for (Descriptors.FieldDescriptor field : wholeMessage.getDescriptorForType().getFields()) {
+                    if (filter.contains(field.getFullName())) {
+                        builder.setField(field, wholeMessage.getField(field));
+                    }
+                }
+
+                filtered.add((M) builder.build());
+            }
+
+        } catch (NoSuchMethodException | InvocationTargetException | IllegalAccessException | InstantiationException e) {
+            // If any reflection failure happens, return all the data without any mask applied.
```

Comment:
Also it is an important part of behaviour description. Should also go to the method JavaDoc section.